View/Elements/tab.ctp

```diff
@@ -1,91 +1,102 @@
-                                    <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="weekly">Weekly Basis</a></li>
-                                    <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="monthly">Monthly Basis</a></li>
-                                    <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="yearly">Yearly Basis</a></li>
-                                </ul>
-                            </div>
-                        <?php else: ?>
-                            <button type="button" class="btn btn-default groupDataByDate" data-value="daily">Filter</button>
-                        <?php endif; ?>
-                    </div>
-                <?php endif; ?>
-                <div class="dataTableContainer" id="<?php print $prefix; ?>DatatableContainer"></div>
-            </div>
-            <!-- /.box-body -->
+          if($this->Form->isFieldError('date' . ucfirst($prefix) . 'Select')) {
+            print $this->Form->error('date' . ucfirst($prefix) . 'Select');
+          }
+          ?>
         </div>
+        <div class="columnChart" id="<?php print Inflector::pluralize($prefix); ?>ChartDetail"></div>
+      </div>
+    <?php endif; ?>
+    <div class="box" data-type="<?php print $prefix ?>">
+      <div class="box-header with-border">
+        <h3 class="box-title"><?php print _txt('pl.rciamstatsviewer.' . $prefix . '.pl'); ?> </h3>
+      </div>
+      <!-- /.box-header -->
+      <div class="box-body dataTableWithFilter">
+        <?php if($vv_permissions['registered']) : ?>
+          <div class="dataTableDateFilter bg-box-silver">
+            From: &nbsp;<input type="text" id="<?php print $prefix ?>DateFrom" name="<?php print $prefix ?>DateFrom" data-provide="datepicker" />
+            &nbsp;&nbsp;&nbsp;To: &nbsp;<input type="text" id="<?php print $prefix ?>DateTo" name="<?php print $prefix ?>DateTo" data-provide="datepicker" />
+            &nbsp;
+            <?php if($prefix == 'registered' || $prefix == 'cou') : ?>
+              <div class="btn-group">
+                &nbsp;<button type="button" class="btn btn-default dropdown-toggle filter-button" data-toggle="dropdown">
+                  Filter <span class="caret"></span>
+                </button>
+                <ul class="dropdown-menu">
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="daily">Daily Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="weekly">Weekly Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="monthly">Monthly Basis</a></li>
+                  <li><a href="#" onclick="return false;" class="groupDataByDate" data-value="yearly">Yearly Basis</a></li>
+                </ul>
```

Comment:
same as above.
I noticed that you use the same Dropdown menu multiple times. Perhaps we should move this into its own element

Reply:
Done, I have put it to an element also